src/app/server/OnboardingCodesUtil.cpp

```diff
@@ -37,29 +37,29 @@ using namespace ::chip::DeviceLayer;
 
 void PrintOnboardingCodes(chip::RendezvousInformationFlags aRendezvousFlags)
 {
-    chip::SetupPayload payload;
+    chip::PayloadContents payload;
 
-    CHIP_ERROR err = GetSetupPayload(payload, aRendezvousFlags);
+    CHIP_ERROR err = GetPayloadContents(payload, aRendezvousFlags);
     if (err != CHIP_NO_ERROR)
     {
-        ChipLogError(AppServer, "GetSetupPayload() failed: %s", chip::ErrorStr(err));
+        ChipLogError(AppServer, "GetPayloadContents() failed: %s", chip::ErrorStr(err));
     }
 
     PrintOnboardingCodes(payload);
 }
 
-void PrintOnboardingCodes(const chip::SetupPayload & payload)
+void PrintOnboardingCodes(const chip::PayloadContents & payload)
 {
-    std::string qrCode;
-    std::string manualPairingCode;
+    char payloadBuffer[chip::QRCodeBasicSetupPayloadGenerator::kMaxQRCodeBase38RepresentationLength];
+    chip::MutableCharSpan qrCode(payloadBuffer);
 
     if (GetQRCode(qrCode, payload) == CHIP_NO_ERROR)
     {
         chip::Platform::ScopedMemoryBuffer<char> qrCodeBuffer;
         const size_t qrCodeBufferMaxSize = strlen(kQrCodeBaseUrl) + strlen(kUrlDataAssignmentPhrase) + 3 * qrCode.size() + 1;
         qrCodeBuffer.Alloc(qrCodeBufferMaxSize);
 
-        ChipLogProgress(AppServer, "SetupQRCode: [%s]", qrCode.c_str());
+        ChipLogProgress(AppServer, "SetupQRCode: [%s]", qrCode.data());
```

Comment:
How do we know it's null-terminated? You may need to use "%.*s" formatting

Reply:
After the most recent changes data() should always be null terminated.